src/devices_models/devices/hydro_generation.jl

```diff
@@ -315,101 +315,55 @@ function cost_function(psi_container::PSIContainer,
-        if use_forecast_data
-            ts_vector = TS.values(PSY.get_data(PSY.get_forecast(PSY.Deterministic,
-                                                                device,
-                                                                initial_time,
-                                                                "storagecapacity")))
-        else
-            ts_vector = ones(time_steps[end])
-        end
-        budget_data[ix] = (name, bus_number, energy_capacity, ts_vector)
-    end
-    return budget_data
-end
 
-function budget_constraints!(psi_container::PSIContainer,
+function energylimit_constraints!(psi_container::PSIContainer,
                     devices::IS.FlattenIteratorWrapper{H},
-                    model::DeviceModel{H, <:AbstractHydroDispatchFormulation},
+                    model::DeviceModel{H, <:AbstractHydroFormulation},
                     system_formulation::Type{<:PM.AbstractPowerModel},
                     feed_forward::Union{Nothing, AbstractAffectFeedForward}) where H<:PSY.HydroGen
-    parameters = model_has_parameters(psi_container)
-    budget_data  = _get_budget(psi_container, devices)
-    if parameters
-        device_budget_param_ub(psi_container,
-                            budget_data,
-                            Symbol("budget_$(H)"), # TODO: better name for this constraint
-                            UpdateRef{H}(:storagecapacity),
-                            Symbol("P_$(H)"))
-    else
-        device_budget_param_ub(psi_container,
-                            budget_data,
-                            Symbol("budget_$(H)"), # TODO: better name for this constraint
-                            Symbol("P_$(H)"))
-    end
+
+    return
 end
 
-function device_budget_param_ub(psi_container::PSIContainer,
-                            budget_data::Vector{Tuple{String, Int64, Float64, Vector{Float64}}},
-                            cons_name::Symbol,
-                            param_reference::UpdateRef,
-                            var_name::Symbol)
-    time_steps = model_time_steps(psi_container)
-    variable = get_variable(psi_container, var_name)
-    set_name = (r[1] for r in budget_data)
-    no_of_budgets = length(budget_data[1][4])
-    time_lengths = time_steps/length(budget_data[1][4])
-    time_chunks = reshape(time_steps, (time_lengths, no_of_budgets))
-    constraint = _add_cons_container!(psi_container, cons_name, set_name, no_of_budgets)
-    param = _add_param_container!(psi_container, param_reference, names, no_of_budgets)
-
-    for data in budget_data, i in 1:no_of_budgets
-        name = data[1]
-        forecast = data[4][i]
-        multiplier = data[3]
-        param[name] = PJ.add_parameter(psi_container.JuMPmodel, forecast)
-        constraint[name] = JuMP.@constraint(psi_container.JuMPmodel,
-                    sum([variable[name, t] for t in time_chunks[:, i]]) <= multiplier*param[name])
+function energylimit_constraints!(psi_container::PSIContainer,
```

Comment:
This function should be implemented regardless of the feedforward, why are you passing the feedforward here?

Reply:
- It's a feed-forward constraint no ? At least for this use case where you decide in the 1st stage what the optimal use of water/energy is and then feed forward the this limit to the 2nd stage.
- I dont know the reason you pass the feedforward model in all the constraint func but I assumed that your intention was to do something like this
- The constraint that should be the default is tracking water/energy in the reservoir 
    - <img src="https://render.githubusercontent.com/render/math?math=e^{stored}_{t-1}-e^{use}_t%2Be^{inflow}_t=e^{stored}_t ">
- If you think the energy limit needs to be a separate constraint regardless of feed-forward we can do it.